tests/tests_suite_test.go

```diff
@@ -125,18 +154,95 @@ func getSsp() *sspv1alpha1.SSP {
 }
 
 func waitUntilDeployed() {
-	Eventually(func() bool {
+	if deploymentTimedOut {
+		Fail("Timed out waiting for SSP to be in phase Deployed.")
+	}
+
+	// Set to true before waiting. In case Eventually fails,
+	// it will panic and the deploymentTimedOut will be left true
+	deploymentTimedOut = true
+	defer func() {
+		if deploymentTimedOut {
+			logDeploymentState()
+		}
+	}()
+
+	EventuallyWithOffset(1, func() bool {
 		return getSsp().Status.Phase == lifecycleapi.PhaseDeployed
 	}, timeout, time.Second).Should(BeTrue())
+	deploymentTimedOut = false
 }
 
 func waitForDeletion(key client.ObjectKey, obj runtime.Object) {
-	Eventually(func() bool {
+	EventuallyWithOffset(1, func() bool {
 		err := apiClient.Get(ctx, key, obj)
 		return errors.IsNotFound(err)
 	}, timeout, time.Second).Should(BeTrue())
 }
 
+func logOperatorState() {
```

Comment:
I think these debug functions are not necessary, because when a job fails, you can find all resources on the cluster under the `artifacts` link in the job page, this includes all resources (equivalent of `oc get $resource --all -A -oyaml`) and all container logs.
So unless there is a reason I am missing here, these functions can be removed

Reply:
You can see all resources on the cluster here (from this PRs job page): https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/kubevirt_ssp-operator/41/pull-ci-kubevirt-ssp-operator-master-e2e-functests/1328977540303294464/artifacts/e2e-functests/gather-extra/

Small enough logs will show up in the browser, the larger ones are downloaded to your machine and are actually `gzip` files, not textual `json` files

Reply:
That is useful. I will remove some of the logging, but probably not all.
The `AfterSuite` part removes the SSP CR and the operator removes all the resources before tests end, so the resources are not in the `artifacts` directory.

Reply:
If the `AfterSuite` removes the SSP CR this is not good.
The test suite should expect an existing deployment (deployed CR) and not remove it at end, so that the test suite can be run downstream as well.

Reply:
It was an oversight on my end, I've forgotten about this requirement during reviews

Reply:
So the `BeforeSuite` should not create a new SSP resource?
We should change this in a new PR.

Reply:
Yes, the `BeforeSuite` should not create a new resources, and the `AfterSuite` should not delete it, a new PR is appropriate